libs/core/langchain_core/vectorstores/base.py

```diff
@@ -307,7 +319,9 @@ async def aadd_documents(
 
             texts = [doc.page_content for doc in documents]
             metadatas = [doc.metadata for doc in documents]
-            return await self.aadd_texts(texts, metadatas, **kwargs)
+            # Extract ids from kwargs to avoid duplication with positional parameter
```

Comment:
Hi @qizwiz thanks for the PR. I don't quite see why this is fixing the issue. Could you add a unit test that reproduces the problem?